owncloudApp/src/main/java/com/owncloud/android/files/services/FileDownloader.java

```diff
@@ -550,11 +535,24 @@ private void notifyDownloadStart(DownloadFileOperation download) {
         showDetailsIntent.putExtra(FileActivity.EXTRA_ACCOUNT, download.getAccount());
         showDetailsIntent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
 
-        mNotificationBuilder.setContentIntent(PendingIntent.getActivity(
-                this, (int) System.currentTimeMillis(), showDetailsIntent, 0
-        ));
-
-        getNotificationManager().notify(R.string.downloader_download_in_progress_ticker, mNotificationBuilder.build());
+        /// create status notification with a progress bar
+        mLastPercent = 0;
+        getNotificationBuilder()
```

Comment:
This could be exported into its own method.

Reply:
Sorry, I don't get what you mean, could you explain it to me?

Reply:
the building of the notification consists of many lines of code that belong together. Pleas export the building into its own methode.

Reply:
I agree that notifications need a rework, but maybe it is not the right moment. I just moved some old code. I don't think that refactor every single file on each PR is the proper way to fix a located problem.

We have a new architecture issue about downloading a file and probably we will refactor or even delete this file.

Reply:
all right.